### PR TITLE
Add Xbox controller support.

### PR DESCRIPTION
**Pull request by boarder2**
This PR adds support for displaying Xbox controller battery status.

It uses https://github.com/nikvoronin/XInput.Wrapper which is MIT licensed.

### CODE REVIEW

**Comment by BarRaider** [comment]
Definitely not going to add the entire source code of XInput library into this plugin. Please use nuget to get that libeary and remove all the redundant code

**Comment by boarder2** [comment]
Good call, updated to use XInput from nuget!

**Comment by boarder2** [comment]
@BarRaider any chance at getting this in an official release?

File: streamdeck-battery.sln [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.29509.3
+# Visual Studio Version 17
```
Comment (BarRaider) [inline streamdeck-battery.sln:3]: Please remove the sln from your PR
No need to change the configuration settings

File: streamdeck-battery/Actions/XboxControllerBatteryStatsAction.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        #region Private Methods
+
+        private Task SaveSettings()
+        {
+            return Connection.SetSettingsAsync(JObject.FromObject(settings));
+        }
+
+        private void StreamDeckConnection_OnPropertyInspectorDidAppear(object sender, streamdeck_client_csharp.StreamDeckEventReceivedEventArgs<streamdeck_client_csharp.Events.PropertyInspectorDidAppearEvent> e)
+        {
+            if (X.IsAvailable)
```
Comment (BarRaider) [inline streamdeck-battery/Actions/XboxControllerBatteryStatsAction.cs:176]: set settings.Devices to null here before checking of X is available

Comment (BarRaider) [reply streamdeck-battery/Actions/XboxControllerBatteryStatsAction.cs:176]: Another option (probably better), given this is just a static list, to put it as a static list of options in the html file
See here as an example:
https://github.com/BarRaider/streamdeck-wintools/blob/c7e951fd8368fb38f363a547c09629f3074c0060/streamdeck-wintools/PropertyInspector/AppAudioMixer.html#L20

File: streamdeck-battery/PropertyInspector/XboxBattery.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <script src="XboxBattery.js"></script>
+</head>
+<body>
+    <div class="sdpi-wrapper">
+        <details class="message">
+            <summary>For support/feedback/suggestions contact me at <span class="linkspan" onclick="openWebsite()">https://BarRaider.com</span> </summary>
+        </details>
+        <div class="sdpi-item" id="dvDevices">
+            <div class="sdpi-item-label">Device</div>
+            <select class="sdpi-item-value select sdProperty sdList" id="devices" oninput="setSettings()" sdListTextProperty="name" sdListValueProperty="name" sdValueField="device"></select>
```
Comment (BarRaider) [inline streamdeck-battery/PropertyInspector/XboxBattery.html:20]: Consider replacing with a static list of 4 entries as per:
https://github.com/BarRaider/streamdeck-wintools/blob/c7e951fd8368fb38f363a547c09629f3074c0060/streamdeck-wintools/PropertyInspector/AppAudioMixer.html#L20

**Review by BarRaider — CHANGES_REQUESTED** [review]

**Comment by BarRaider** [comment]
Sorry, forgot about this - made some change requests, nothing major - as soon as completed I'll merge to mainline

File: streamdeck-battery.sln (hunk 1 above)
Comment (boarder2) [reply streamdeck-battery.sln]: Removed the sln changes from the PR

File: streamdeck-battery/Actions/XboxControllerBatteryStatsAction.cs (hunk 2 above)
Comment (boarder2) [reply streamdeck-battery/Actions/XboxControllerBatteryStatsAction.cs]: Updated to a static list in html

File: streamdeck-battery/PropertyInspector/XboxBattery.html (hunk 3 above)
Comment (boarder2) [reply streamdeck-battery/PropertyInspector/XboxBattery.html]: Updated to a static list in html

**Comment by boarder2** [comment]
Should be good to go now!

**Comment by BarRaider** [comment]
Thanks for your contribution. Merging into mainline and will deploy a version on Discord very soon (https://buz.bz/d)

**Pull request merged**